browser/ui/views/brave_shields/cookie_consent_browser_helper.h

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_BRAVE_SHIELDS_COOKIE_CONSENT_BROWSER_HELPER_H_
+#define BRAVE_BROWSER_UI_VIEWS_BRAVE_SHIELDS_COOKIE_CONSENT_BROWSER_HELPER_H_
+
+#include <memory>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "base/scoped_observation.h"
+#include "chrome/browser/ui/browser_user_data.h"
+#include "chrome/browser/ui/tabs/tab_strip_model_observer.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/widget/widget_observer.h"
+#include "url/gurl.h"
+
+class CookieConsentBlockerUI;
+
+template <typename T>
+class WebUIBubbleManagerT;
+
+namespace content {
+class WebContents;
+}
+
+namespace brave_shields {
+
+class CookieConsentBrowserHelper
```

Comment:
I think this might be cleaner to implement as a NavigationThrottle (using the WebContents from the NavigationHandle to see if it's loading in the active tab). Also I believe the upstream convention for naming for these is *BubbleHost, but with a navigation throttle you'd split it into two classes `*NavigationThrottle` and `*BubbleHost`